python/resdata/resfile/rd_kw.py

```diff
@@ -765,25 +752,21 @@ def assign(self, value, mask=None, force_active=False):
             if type(value) == type(self):
                 if mask is not None:
                     mask.copy_kw(self, value, force_active)
+                elif self.assert_binary(value):
```

Comment:
`self.assert_binary` in the tests is never false, we should add a test for that case.